src/NUnitEngine/nunit.engine/Services/RuntimeFrameworkService.cs

```diff
@@ -72,6 +75,23 @@ private static bool FrameworksMatch(RuntimeFramework f1, RuntimeFramework f2)
                    f1.FrameworkVersion.Minor == f2.FrameworkVersion.Minor;
         }
 
+        private static bool RuntimesMatch(RuntimeType rt1, RuntimeType rt2)
+        {
+            if (rt1 == rt2)
+                return true;
+
+            if (rt1 == RuntimeType.Any || rt2 == RuntimeType.Any)
+                return true;
+
+            if (rt1 == RuntimeType.Net && rt2 == RuntimeType.Mono)
+                return true;
+
+            if (rt1 == RuntimeType.Mono && rt2 == RuntimeType.Net)
+                return true;
```

Comment:
Can you clarify what this method is doing, sorry? It makes me a little uneasy that we have a method called `RuntimesMatch` which returns true for two different runtimes: netfx and mono. Maybe its just a naming issue?

Reply:
I totally agree. I left it alone because it's not new and my additions were in the same spirit as the original... where "match" didn't mean "equals" IMO we need to get back to this soon and redo the whole set of runtime classes...

However, I can only work on one thing at a time. :confused:

Reply:
Again, if we're happy this doesn't break any existing scenarios, then let's go with it and get this all in. 🙂 

The `RuntimesMatch` method name is new from what I can see, however. Could we call this something else? Is it another `RuntimeSupports` maybe?

Reply:
I believe it doesn't break anything... and all the same tests are still there!

Longer story... in the GUI fork, Runtime is a class, which knows how to compare itself to other runtimes. In this engine, it's still just an enum. So created this method to match the same functionality, but in a procedural way so as not to add unrelated work (redesigning RuntimeFramework) to the PR. I've done that a lot here, and I suppose it's a little confusing if you don't know where it comes from.

RuntimeSupports captures the meaning perfectly.

Reply:
OK... this is a can of works and you should review the changes to this file again.

I left the name the same and changed the names of the arguments. As used, the first is always the requested runtime and the second is the available runtime. The available one can never actually be Any, so I eliminated that. So far so good.

This method is called by the FrameworksMatch method so I made the same change to that one. But that method also compares versions, using CLR version for some things, framework version for others. That's because, back in the day, we supported requesting a _specific_ CLR version, i.e. pre-releases of .NET 2.0 and .NET 4.0. We don't need that any longer and I'd like to get away from making much use of CLR version if possible, so I changed the method so that all checks now use FrameworkVersion. All tests still pass, so I feel safe, but you should still look at it.

As I reviewed the code, I realized that this support check is actually implemented in both `RuntimeFramework` (`Supports` method) and in the `RuntimeFrameworkService`. It's possible that I was in the process of moving it from `RuntimeFramework` to the service at the point in time when I had to stop working on the engine. I vaguely remember that was my intent anyway. We should get back to this.

Reply:
Thanks Charlie. This is indeed a can of worms. 😄 

The changes look good, and your explanation makes sense. I think there are two duplicate lines that are no longer needed - I've commented on those. 

The only thing that still makes me uneasy is the name "RuntimesMatch" for a method which returns true for two different runtimes. Likewise for "FrameworksMatch" to be honest. That said, this whole area really needs work, so if you'd prefer to leave this for now, I'm happy to go with that.